### PR TITLE
Implement select new competitor

### PR DESCRIPTION
**Pull request by duncanspumpkin**
not tested

### CODE REVIEW

File: src/OpenLoco/src/World/CompanyManager.cpp [hunk 1]
```diff
@@ -413,6 +414,7 @@ namespace OpenLoco::CompanyManager
             if (metric < bestInstalledValue)
             {
                 bestInstalled = std::vector<ObjectHeader>{ *installed.second._header };
+                bestInstalledValue = metric;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/World/CompanyManager.cpp:417]: Does this warrant a changelog entry?

Comment (duncanspumpkin) [reply src/OpenLoco/src/World/CompanyManager.cpp:417]: What do you mean

Comment (AaronVanGeffen) [reply src/OpenLoco/src/World/CompanyManager.cpp:417]: Oh, the implication in the commit message was that the value was wrong on master, no?

Comment (duncanspumpkin) [reply src/OpenLoco/src/World/CompanyManager.cpp:417]: oh right that was an implementation mistake.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/World/CompanyManager.cpp:417]: Ahh, I see now that code in question was introduced in the first commit. Never mind, then!

**Review by AaronVanGeffen — COMMENTED** [review]
This branch crashes for me when I start a new scenario (Boulder Breakers) with no preferred company face or name set. It works fine when it is set.

Screenshot 2024-08-25 at 22 18 47

**Review by AaronVanGeffen — APPROVED** [review]
Works fine now! Assuming the fix had no bearing on replays, given it only affects new games.

**Comment by duncanspumpkin** [comment]
> Works fine now! Assuming the fix had no bearing on replays, given it only affects new games.

Ai can actually generate new companies and use this code during a game so its not just new games. But it doesn't appear to have any replay issues ... (now).

**Pull request merged**